### PR TITLE
Change checkout behaviour

### LINKED ISSUE #13126: Change checkout behavior
**Describe the bug**
#12456 was not done completely
Here is what remains to be done:

The main difference vs current behaviour is that the user will always have to click on "Continue" to go from Step X to Step X+1, no matter if :

- he/she goes in the different steps for the first time
- he/she went to the step X and decided to go back to X-1 or X-2 to check or modify the information stored.

This means that :

- The next steps after the current step are no longer clickable (only Continue button allows to go to the next step)
- There should be a "Continue" button in the Personal information step (in case the logged in user goes back to the 1st step)
- Green checks and "Modify" CTAs should be removed in the steps following the user current one.

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Change checkout behavior, next steps are not accessible any more, and continue button must be clicked each time
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #13126
| How to test?  | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13440)

### CODE REVIEW

File: themes/_core/js/checkout.js [hunk 1]
```diff
@@ -41,25 +41,20 @@ function handleCheckoutStepChange() {
...
-    }
-  );
+        $clickedStep.toggleClass('-current');
+        $clickedStep.toggleClass(currentStepClass);
 
-  $(currentStepSelector + ':not(#checkout-personal-information-step)').nextAll().on(
-    'click',
-    (event) => {
-      $(currentStepSelector + ' button.continue').click();
+        let $nextSteps = $clickedStep.nextAll();
```
Comment (PierreRambaud) [inline themes/_core/js/checkout.js]:
```suggestion
        const $nextSteps = $clickedStep.nextAll();
```

File: themes/_core/js/checkout.js [hunk 2]
```diff
@@ -41,25 +41,20 @@ function handleCheckoutStepChange() {
...
   let currentStepSelector = '.' + currentStepClass;
-  let stepsAfterPersonalInformation = $('#checkout-personal-information-step').nextAll();
 
-  $(currentStepSelector).prevAll().add(stepsAfterPersonalInformation).on(
+  $(currentStepSelector).prevAll().on(
     'click',
     (event) => {
-      let $nextStep = $(event.target).closest('.checkout-step');
-      if (!$nextStep.hasClass('-unreachable')) {
+      let $clickedStep = $(event.target).closest('.checkout-step');
```
Comment (PierreRambaud) [inline themes/_core/js/checkout.js]:
```suggestion
      const $clickedStep = $(event.target).closest('.checkout-step');
```

Comment (jolelievre) [reply themes/_core/js/checkout.js]: Done! (it's funny it does not change the built js..) It must be converted back to a `var`

Comment (PierreRambaud) [reply themes/_core/js/checkout.js]: yep :)

**Review by PierreRambaud — APPROVED** [review]

**Review by jolelievre — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13440)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @marionf and @PierreRambaud)

___
*[themes/_core/js/checkout.js, line 54 at r1](https://reviewable.io/reviews/prestashop/prestashop/13440#-LcamlrSKSwKVXXCKMj--r1-54:-LdisYHkAUm6PSDk9Kzr:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/0cd5a3dc366e3aa73a57b2674665ff4cf9b9b1ca/themes/_core/js/checkout.js#L54)):*
<details><summary><i>Previously, PierreRambaud (GoT) wrote…</i></summary><blockquote>

```suggestion
        const $nextSteps = $clickedStep.nextAll();
```
</blockquote></details>

Done.

File: themes/classic/templates/checkout/_partials/steps/personal-information.tpl [hunk 3]
```diff
@@ -32,6 +32,20 @@
...
+          name="controller"
+          type="submit"
+          value="order"
+        >
+          {l s='Continue' d='Shop.Theme.Actions'}
+        </button>
+      </form>
+
+    </div>
+
```
Comment (rdy4ever) [inline themes/classic/templates/checkout/_partials/steps/personal-information.tpl:48]: One little BC problem here: when using a theme that doesn't include this button, as a logged-in user,  (from add-ons or other custom theme), if the user clicks on step one from another step (after step 1 is completed), then there's no way for him to go back (or forward, te be exact) to the next step, as this button is missing and he cannot click on the other steps. So he gets stuck and needs to refresh the page.

**Pull request merged**